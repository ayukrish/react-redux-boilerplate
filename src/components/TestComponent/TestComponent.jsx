import React from 'react';
import PropTypes from 'prop-types';
import ErrorBoundary from '../ErrorBoundary/ErrorBoundary';

const propTypes = {
  value: PropTypes.string,
  onChange: PropTypes.func
};

const defaultProps = {
  value: '',
  onChange: () => { }
};

const TestComponent = (props) => {
  const {
    value,
    onChange
  } = props;
  return (
    <ErrorBoundary>
      <input
        type="text"
        value={value}
        onChange={(e) => onChange(e.target.value)}
      />
      <div>{value}</div>
    </ErrorBoundary>
  );
};

TestComponent.propTypes = propTypes;
TestComponent.defaultProps = defaultProps;

export default TestComponent;
