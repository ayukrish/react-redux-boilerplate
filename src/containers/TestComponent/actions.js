import { ON_CHANGE } from './constants';


export const onChange = (data) => (
  {
    type: ON_CHANGE,
    data
  }
);
