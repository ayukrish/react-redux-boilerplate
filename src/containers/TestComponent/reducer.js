import { ON_CHANGE } from './constants';

const INITIAL_STATE = {
  data: ''
};

/**
  * Reducer - to update services state
  * @param {Object} state initial/default services state.
  */
export default function (state = INITIAL_STATE, { type, data }) {
  switch (type) {
  case ON_CHANGE:
    return { ...state, data };
  default:
    return state;
  }
}
