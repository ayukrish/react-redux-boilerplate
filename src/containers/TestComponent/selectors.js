export const valueData = state => (state.value && state.value.data) || '';
