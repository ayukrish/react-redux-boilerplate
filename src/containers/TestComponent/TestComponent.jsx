/** eslint-disable */
import { connect } from 'react-redux';
// import { createStructuredSelector } from 'reselect';
import { onChange } from './actions';
import TestComponent from '../../components/TestComponent/TestComponent';
// import { valueData } from './selectors';

// const mapStateToProps = createStructuredSelector({
//   value: valueData,
// });

const mapStateToProps = (state) => ({
  value: (state.value && state.value.data) || '',
});

const mapDispatchToProps = (dispatch) => ({
  onChange: (data) => dispatch(onChange(data)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(TestComponent);
