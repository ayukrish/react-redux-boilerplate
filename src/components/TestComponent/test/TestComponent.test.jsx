import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

// Components
import TestComponent from '../TestComponent';


describe('TestComponent Test Suite', () => {
  it('renders correctly', () => {
    const onChange = jest.fn();
    const output = shallow(
      <TestComponent
        onChange={onChange}
      />,
    );
    expect(shallowToJson(output)).toMatchSnapshot();
    output.find('input').simulate('change', { target: {} });
    expect(onChange).toBeCalled();
  });

  it('should have default handleChange', () => {
    expect(TestComponent.defaultProps.onChange()).toBe(undefined);
  });
});
