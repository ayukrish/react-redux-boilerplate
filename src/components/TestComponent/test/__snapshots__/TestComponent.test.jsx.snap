// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent Test Suite renders correctly 1`] = `
<ErrorBoundary>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <div />
</ErrorBoundary>
`;
